testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2026,39 +2028,34 @@ public void testCommentTable()
 
             // comment updated
             assertUpdate("COMMENT ON TABLE " + table.getName() + " IS 'updated comment'");
-            assertThat(getTableComment(table.getName())).isEqualTo("updated comment");
+            assertThat(getTableComment(catalogName, schemaName, table.getName())).isEqualTo("updated comment");
 
             // comment set to empty or deleted
             assertUpdate("COMMENT ON TABLE " + table.getName() + " IS ''");
-            assertThat(getTableComment(table.getName())).isIn("", null); // Some storages do not preserve empty comment
+            assertThat(getTableComment(catalogName, schemaName, table.getName())).isIn("", null); // Some storages do not preserve empty comment
 
             // comment deleted
             assertUpdate("COMMENT ON TABLE " + table.getName() + " IS 'a comment'");
-            assertThat(getTableComment(table.getName())).isEqualTo("a comment");
+            assertThat(getTableComment(catalogName, schemaName, table.getName())).isEqualTo("a comment");
             assertUpdate("COMMENT ON TABLE " + table.getName() + " IS NULL");
-            assertThat(getTableComment(table.getName())).isEqualTo(null);
+            assertThat(getTableComment(catalogName, schemaName, table.getName())).isEqualTo(null);
         }
 
         String tableName = "test_comment_" + randomTableSuffix();
         try {
             // comment set when creating a table
             assertUpdate("CREATE TABLE " + tableName + "(key integer) COMMENT 'new table comment'");
-            assertThat(getTableComment(tableName)).isEqualTo("new table comment");
+            assertThat(getTableComment(catalogName, schemaName, tableName)).isEqualTo("new table comment");
         }
         finally {
             assertUpdate("DROP TABLE IF EXISTS " + tableName);
         }
     }
 
-    private String getTableComment(String tableName)
+    private String getTableComment(String catalogName, String schemaName, String tableName)
     {
-        // TODO use information_schema.tables.table_comment
-        String result = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
-        Matcher matcher = Pattern.compile("COMMENT '([^']*)'").matcher(result);
-        if (matcher.find()) {
-            return matcher.group(1);
-        }
-        return null;
+        String sql = format("SELECT comment FROM system.metadata.table_comments WHERE catalog_name = '%s' AND schema_name = '%s' AND table_name = '%s'", catalogName, schemaName, tableName);
```

Comment:
Thank you. Much better.

Are there any connectors where the predicates won't get pushed down though? e.g. maybe Hive? cc: @findepi ?

Reply:
@hashhar not sure what's the question. `system.metadata.table_comments` is engine-provided and predicate pushdown happens there in a generic fashion.

Reply:
Nevermind, I see SHOW CREATE TABLE is rewritten to fetch table metadata + column props + column metadata anyway.